flybywiresim/aircraft
# feat(EFB/SimBrief): Automatically import SimBrief Fuel & Payload

Fixes #7772
(Together with PR #8407)

## Summary of Changes


- When SimBrief Data was imported (either manual or automatic), the Fuel & Payload Pages automatically import their Data too. Approach:
  - The Fuel and Payload Page have an useEffect to import the Data when visited (only when SimBrief Data is loaded)
  - The Import Function navigates through the Pages (Dashboard->Fuel->Payload->Dashboard)
  - Two additional Flags in the SimBrief Store prevent that Payload or Fuel are imported every time the respective Page is revisited
- Auto-Import of SimBrief Data moved from EFB.tsx to FlightWidget.tsx (to not maintain two Copies of the same Code)
- Added Setting to enable/disable that Behavior

## Screenshots (if necessary)


![flightwidget-toast](https://github.com/flybywiresim/aircraft/assets/78614902/1ff0c9fe-92f0-422f-924c-558cfdf1a537)

![settings-atsu](https://github.com/flybywiresim/aircraft/assets/78614902/347724c6-910d-4d8a-9ca1-958ed24f8e62)


## References





## Additional context

Although the originating Issue is in the GSX Context, this Feature improves Comfort for everyone (using SimBrief). Only one Click needed - if at all - to get all relevant Data imported 😃 


Discord username (if different from GitHub): Fragtality

## Testing instructions


Test 1

1. Prepare a Flightplan on SimBrief for the FBW
2. Load into the Plane and enable the new Setting in the EFB (Settings > ATSU/AOC). Ensure that your SimBrief Account is set.
3. Navigate to the Dashboard and hit "Import SimBrief Data" => A Notification should appear (like shown in the Screenshots)
4. Navigate to the Fuel & Payload Page => Each Page should display the planned Fuel / Payload from your SimBrief FP

Test 2

1. If you don't use the (already existing) Auto-Import Setting: Please enable it
2. Poweroff the EFB
3. Power it on again => You should see the same Effects as before (Notification, Fuel & Payload Page have the planned Values)

For additional Verification you can monitor the L-Vars `A32NX_FUEL_DESIRED` and `A32NX_AIRFRAME_ZFW_DESIRED` => these should immediately Change when the SimBrief Data was imported (without leaving the Dashboard).
(If you have a Tool at Hand to monitor L-Vars quick & easy)


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
